src/org/ods/orchestration/scheduler/LeVADocumentScheduler.groovy

```diff
@@ -81,67 +81,67 @@ class LeVADocumentScheduler extends DocGenScheduler {
         ]
     ]
 
+    // Document types per repository type with an optional phase constraint
+    private final static Map REPSITORY_TYPES = [
+        (MROPipelineUtil.PipelineConfig.REPO_TYPE_ODS_CODE): [
+            (LeVADocumentUseCase.DocumentType.DTR as String): null,
+            (LeVADocumentUseCase.DocumentType.TIR as String): null
+        ],
+        (MROPipelineUtil.PipelineConfig.REPO_TYPE_ODS_INFRA): [
+            (LeVADocumentUseCase.DocumentType.TIR as String): null
+        ],
+        (MROPipelineUtil.PipelineConfig.REPO_TYPE_ODS_SERVICE): [
+            (LeVADocumentUseCase.DocumentType.TIR as String): null
+        ],
+        (MROPipelineUtil.PipelineConfig.REPO_TYPE_ODS_TEST): [:],
+    ]
+
     // Document types per pipeline phase with an optional lifecycle constraint
-    private static Map PIPELINE_PHASES = [
+    private final static Map PIPELINE_PHASES = [
         (MROPipelineUtil.PipelinePhases.INIT): [
             (LeVADocumentUseCase.DocumentType.CSD as String): MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_END
         ],
         (MROPipelineUtil.PipelinePhases.BUILD): [
             (LeVADocumentUseCase.DocumentType.DTP as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
+            (LeVADocumentUseCase.DocumentType.TIP as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
+            (LeVADocumentUseCase.DocumentType.RA as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
+            (LeVADocumentUseCase.DocumentType.IVP as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
+            (LeVADocumentUseCase.DocumentType.CFTP as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
+            (LeVADocumentUseCase.DocumentType.TCP as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
             (LeVADocumentUseCase.DocumentType.DTR as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_EXECUTE_REPO,
             (LeVADocumentUseCase.DocumentType.OVERALL_DTR as String): MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_END
         ],
         (MROPipelineUtil.PipelinePhases.DEPLOY): [
-            (LeVADocumentUseCase.DocumentType.TIP as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_START,
             (LeVADocumentUseCase.DocumentType.TIR as String): MROPipelineUtil.PipelinePhaseLifecycleStage.POST_EXECUTE_REPO
         ],
         (MROPipelineUtil.PipelinePhases.TEST): [
             (LeVADocumentUseCase.DocumentType.SSDS as String): MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_END,
```

Comment:
@s2oBCN also SSDS needs to be created in the Build phase

Reply:
Yes, I tested it in the Build phase but failed because it needs a file, that it isn't at that phase

Reply:
@s2oBCN thanks for clarifying, your reasoning in our conversation is spot-on: in 3.x, the SSDS depends on SonarQube data.